cajita/unit_test/tstHypreSemiStructuredSolverMulti.hpp

```diff
@@ -0,0 +1,297 @@
+#include <Cajita_GlobalMesh.hpp>
+#include <Cajita_HypreSemiStructuredSolver.hpp>
+#include <Cajita_HypreStructuredSolver.hpp>
+#include <Cajita_Partitioner.hpp>
+#include <Cajita_ReferenceStructuredSolver.hpp>
+#include <Cajita_Types.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <gtest/gtest.h>
+
+#include <array>
+#include <vector>
+
+using namespace Cajita;
+
+namespace Test
+{
+
+//---------------------------------------------------------------------------//
+// FIXME: Only run test if HYPRE is compatible with the memory space. This
+// is currently written in this structure because HYPRE only has
+// compile-time switches for backends and hence only one can be used at a
+// time. Once they have a run-time switch we can use that instead.
+template <class MemorySpace>
+std::enable_if_t<!HypreIsCompatibleWithMemorySpace<MemorySpace>::value, void>
+poissonTest( const std::string&, const std::string&, MemorySpace )
+{
+}
+
+template <class MemorySpace>
+std::enable_if_t<HypreIsCompatibleWithMemorySpace<MemorySpace>::value, void>
+poissonTest( const std::string& solver_type, const std::string& precond_type,
+             MemorySpace )
+{
+    // Create the global grid.
+    double cell_size = 0.25;
+    std::array<bool, 3> is_dim_periodic = { false, false, false };
+    std::array<double, 3> global_low_corner = { -1.0, -2.0, -1.0 };
+    std::array<double, 3> global_high_corner = { 1.0, 1.0, 0.5 };
+    auto global_mesh = createUniformGlobalMesh( global_low_corner,
+                                                global_high_corner, cell_size );
+
+    // Create the global grid.
+    DimBlockPartitioner<3> partitioner;
+    auto global_grid = createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                         is_dim_periodic, partitioner );
+
+    // Create a local grid.
+    auto local_mesh = createLocalGrid( global_grid, 1 );
+    auto owned_space = local_mesh->indexSpace( Own(), Cell(), Local() );
+
+    // Create the RHS.
+    auto vector_layout = createArrayLayout( local_mesh, 3, Cell() );
+    auto rhs = createArray<double, MemorySpace>( "rhs", vector_layout );
+    //    auto rhs = createArray<double, Kokkos::LayoutRight, MemorySpace>(
+    //    "rhs", vector_layout );
+    ArrayOp::assign( *rhs, 1.0, Own() );
+
+    // Create the LHS.
+    auto lhs = createArray<double, MemorySpace>( "lhs", vector_layout );
+    //    auto lhs = createArray<double, Kokkos::LayoutRight, MemorySpace>(
+    //    "lhs", vector_layout );
+    ArrayOp::assign( *lhs, 0.0, Own() );
+
+    HYPRE_Init();
+
+    // Create a solver.
+    auto solver = createHypreSemiStructuredSolver<double, MemorySpace>(
+        solver_type, *vector_layout, false, 3 );
+
+    // Create a 7-point 3d laplacian stencil.
+    std::vector<std::array<int, 3>> stencil = {
+        { 0, 0, 0 }, { -1, 0, 0 }, { 1, 0, 0 }, { 0, -1, 0 },
+        { 0, 1, 0 }, { 0, 0, -1 }, { 0, 0, 1 } };
+
+    solver->createMatrixStencil( 3, 0, 3, { 7, 0, 0 } );
```

Comment:
In general, would the stencil be repeated, i.e. { 7, 7, 7 } for a full vector equation, for each variable (e.g. for var0 = u_x <--> f_xx, f_xy, f_xz, then var1 = u_y <--> f_yx, f_yy, f_yz, and finally, var2 = u_z <--> f_zx, f_zy, f_zz )?

Reply:
Yes, the vector passed to `createMatrixStencil` contains the number of stencil values from the current equation # (the second input integer) to each of the variables in the system